packages/ui-core/src/components/Search/Search.tsx

```diff
@@ -0,0 +1,152 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import { useCallback, useState, useRef } from 'react';
+import debounce from 'lodash.debounce';
+import SelectItem from '../Select/SelectItem';
+import cnCreate from 'utils/cnCreate';
+import './Search.less';
+
+type HandleSearchFieldSubmit = (e?: React.MouseEvent<HTMLDivElement>) => void;
+type HandleSelectSubmit = (e: React.MouseEvent, index: number) => void;
+type HandleSubmit = (index: number) => void;
+
+export interface ISearchProps {
+    /** Selected value. Should correlate with items.value */
+    value?: string;
+    /** Placeholder */
+    placeholder?: string;
+    /** Array of objects to be used for options rendering */
+    items?: Array<{
+        /** Header */
+        title: string;
+        /** Value */
+        value: string;
+    }>;
+    /** Change handler */
+    onChange?: (value: string) => void;
+    /** Submit handler */
+    onSubmit?: (value: string) => void;
+}
+
+const cn = cnCreate('mfui-search');
+const Search: React.FC<ISearchProps> = ({
+        value = '',
+        placeholder,
+        onChange,
+        onSubmit,
+        items = [],
+}) => {
+    const [searchQuery, setSearchQuery] = useState(value);
+    const [activeIndex, setActiveIndex] = useState(-1);
+    const [isFocused, setFocus] = useState(false);
+    const debouncedOnChange = useRef(debounce((inputValue) => onChange && onChange(inputValue), 500));
+
+    const handleChange: React.EventHandler<React.ChangeEvent<HTMLInputElement>> =
+        useCallback((e): void => {
+            const { target: { value: inputValue = '' } = {} } = e;
```

Comment:
```suggestion
            const { target: { value: inputValue = '' } } = e;
```
Тип события гарантирует объект.